rir/src/api.cpp

```diff
@@ -173,9 +173,11 @@ SEXP pirCompile(SEXP what, const std::string& name, pir::DebugOptions debug) {
     // compile to pir
     pir::Module* m = new pir::Module;
     pir::StreamLogger logger(debug);
+    logger.beginPhase("Compiling " + name);
     pir::Rir2PirCompiler cmp(m, logger);
     cmp.compileClosure(what, name,
                        [&](pir::Closure* c) {
+                           logger.endPhase();
```

Comment:
I do not get exactly what is the idea of a phase. Why the phase ends just before pir starts to optimize? Why the optimization is not another phase?

Reply:
this is just `flush()` that I renamed to phase. The idea is to flush all streams after rir2pir, after optimizations and after pir2rir, so you get some output, if it crashes in one of the steps.

should I name it back to flush? I think that was probably the better name